src/Stratis.Bitcoin.Features.Api/NoCacheMiddleware.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+
+namespace Stratis.Bitcoin.Features.Api
+{
+    /// <summary>
+    /// Middleware to set the response Cache-Control to no-cache.
+    /// </summary>
+    public class NoCacheMiddleware
+    {
+        private readonly RequestDelegate next;
+
+        public NoCacheMiddleware(RequestDelegate next)
+        {
+            this.next = next;
+        }
+
+        public async Task InvokeAsync(HttpContext context)
+        {
+            context.Response.Headers["Cache-Control"] = "no-cache";
```

Comment:
Does this apply?:

https://stackoverflow.com/questions/1294337/adding-context-response-headers-addcache-control-no-cache-says-iis-integ

Reply:
That question was asked before Kestrel and .NET Core existed. It also looks like it's setting the same header.

Reply:
Cool beans.

Reply:
see my comment below, btw the problem only happens when using a browser on swagger right? E.g. if you call endpoints from a console app it works fine I suppose, anyway it doesn't harm to put a nocache header on responses